src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -747,6 +747,14 @@ class CORE_EXPORT QgsWmsLegendNode : public QgsLayerTreeModelLegendNode
 
     void invalidateMapBasedData() override;
 
+    /**
+     *  Lazily initializes mImage
```

Comment:
We need get rid of  `mImage` in the documentation, and state that if synchronous is false, the returned image could be invalid, and it would require to wait for getLegendGraphicFinished/Errored/Progress signals.

We also need to make those signals public.

Reply:
Given that the only usage is from server code, would it be more convenient to keep it private add a new method that fetches the image synchronously?

Reply:
done with https://github.com/qgis/QGIS/pull/58790/commits/1f86ad11b7aba9d8e3e48b0c5021e1d75afb1cfc

Reply:
Yes, it's better this way